opensearch/src/main/java/org/opensearch/sql/opensearch/storage/script/filter/lucene/LuceneQuery.java

```diff
@@ -32,10 +34,13 @@
 import org.opensearch.sql.expression.ReferenceExpression;
 import org.opensearch.sql.expression.function.BuiltinFunctionName;
 import org.opensearch.sql.expression.function.FunctionName;
+import org.opensearch.sql.opensearch.data.type.OpenSearchDateType;
 
 /** Lucene query abstraction that builds Lucene query from function expression. */
 public abstract class LuceneQuery {
 
+  private List<DateFormatter> dateFormatters;
```

Comment:
Is it safe to store this list variable here? I think we should refactor method 
```java
ExprValue cast(FunctionExpression castFunction)
```
 to 
```java
ExprValue cast(FunctionExpression castFunction), List<DateFormatter> dateFormatters
```
and remove this variable.

Reply:
Lucene query should not ref to dateFormatters.

Reply:
I refactored this to use available reference expression to identify the `type`. I have moved the formatter logic to `OpenSearchDateType`  class helper function and just passing that here in the cast map.